packages/database/src/core/view/QueryParams.ts

```diff
@@ -289,13 +289,21 @@ export class QueryParams {
   }
 
   startAfter(indexValue: unknown, key?: string | null): QueryParams {
-    let childKey: string;
-    if (key == null) {
-      childKey = MAX_NAME;
+    let params: QueryParams;
+    if (this.index_ === KEY_INDEX) {
+      if (typeof indexValue === 'string') {
```

Comment:
This check makes it so that startAfter still throws exceptions for incorrectly typed keys.

Reply:
Where does it throw? Should we throw at the callsite to preserve the stacktrace of the original mistake?

Reply:
It throws in validateQueryEndpoints_: https://github.com/firebase/firebase-js-sdk/blob/master/packages/database/src/api/Query.ts#L110, which is the same place that startAt() and endAt() throw for this case. 

I think that the way it's done now the original mistake, passing a non-string key to `startAfter()` should still be clear. I'll also update the error message there.